transport/rest/src/main/java/io/a2a/transport/rest/handler/RestHandler.java

```diff
@@ -0,0 +1,433 @@
+                    tube.complete();
+                }
+            });
+        });
+    }
+
+    private int mapErrorToHttpStatus(JSONRPCError error) {
+        if (error instanceof InvalidRequestError || error instanceof InvalidParamsError) {
+            return 400;
+        }
+        if (error instanceof MethodNotFoundError || error instanceof TaskNotFoundError) {
+            return 404;
+        }
+        if (error instanceof TaskNotCancelableError || error instanceof PushNotificationNotSupportedError
+                || error instanceof UnsupportedOperationError) {
+            return 501;
+        }
+        if (error instanceof ContentTypeNotSupportedError) {
+            return 415;
+        }
+        if (error instanceof InternalError || error instanceof InvalidAgentResponseError) {
+            return 500;
+        }
+        return 500;
+    }
+
+    public AgentCard getAgentCard() {
+        return agentCard;
+    }
+
+    public static class HTTPRestResponse {
+
+        private final int statusCode;
+        private final String contentType;
+        private final String body;
+
+        public HTTPRestResponse(int statusCode, String contentType, String body) {
+            this.statusCode = statusCode;
+            this.contentType = contentType;
+            this.body = body;
+        }
+
+        public int getStatusCode() {
+            return statusCode;
+        }
+
+        public String getContentType() {
+            return contentType;
+        }
+
+        public String getBody() {
+            return body;
+        }
+
+        @Override
+        public String toString() {
+            return "HTTPRestResponse{" + "statusCode=" + statusCode + ", contentType=" + contentType + ", body=" + body + '}';
+        }
+    }
+
+    public static class HTTPRestStreamingResponse extends HTTPRestResponse {
+
+        private final Flow.Publisher<String> publisher;
+
+        public HTTPRestStreamingResponse(Flow.Publisher<String> publisher) {
+            super(200, "text/event-stream", null);
+            this.publisher = publisher;
+        }
+
+        public Flow.Publisher<String> getPublisher() {
+            return publisher;
+        }
+    }
+
+    private static class HTTPRestErrorResponse {
+
+        private final String error;
+        private final String message;
+
+        public HTTPRestErrorResponse(String error, String message) {
+            this.error = error;
+            this.message = message;
+        }
+
+        public HTTPRestErrorResponse(JSONRPCError jsonRpcError) {
+            this.error = jsonRpcError.getClass().getName();
+            this.message = jsonRpcError.getMessage();
+        }
+
+        public String getError() {
+            return error;
+        }
+
+        public String getMessage() {
+            return message;
+        }
+
+        public String toJson() {
+            return "{\"error\": \"" + error + "\", \"message\": \"" + message + "\"}";
```

Comment:
Just a heads up, this hasn't been merged yet but once it is we should make sure our response contains a "code" and the error "message"

https://github.com/a2aproject/A2A/pull/976
https://github.com/a2aproject/A2A/pull/976/files#diff-eebad86dc55e949d1d0630c4a2e92fed2a16b19e0ab623719ed429b5d3eee101R1252